### PR TITLE
add missing qidents_of function

### LINKED ISSUE #479: Goto definitions don't always work
Goto definitions don't always work. 

For example: in the file `logic.lp` the goto definition works for `2` but not for `1` (for both emacs and vscode). The LSP server logs show that it is not able to find the definitions in case of `1`.
```
theorem not_elimp : Π (A : Prop), P (not (not A)) → P A
proof
    assume A nnp
    refine or_elim A (not A) A _ _ _
    refine excl_mid A
    assume pa
    refine pa
    assume pna
    apply bot_elim        <------------------------------- 1
    apply not_elim (not A) nnp
    apply pna
qed

set builtin "P"     ≔ P  <------------------------------- 2
set builtin "T"     ≔ T
set builtin "eq"    ≔ eq
set builtin "eqind" ≔ eqind
set builtin "refl"  ≔ refl
```

### PR DESCRIPTION
**Pull request by firewall2142**
`qidents_of_*` functions were not implemented in `pure/cmd_analysis.ml` causing goto defintions to not work.
This fixes #479

### CODE REVIEW

File: src/pure/cmd_analysis.ml [hunk 1]
```diff
@@ -92,10 +135,18 @@ let qidents_of_cmd (cmd : t) =
...
+    let symqidlist = [ident_to_qident p_sym_nam] in
+    let prfqidlist = begin
+      match p_sym_prf with
+      | Some (ptac_list, _) -> List.concat_map qidents_of_p_tactic ptac_list
+      | None -> []
+    end in
     let terms_list = some_or_empty p_sym_typ @ some_or_empty p_sym_trm in
-    filter_bound_qidents p_sym_arg terms_list
+    let res =  (filter_bound_qidents p_sym_arg terms_list) in
+    res @ symqidlist @ prfqidlist
```
Comment (fblanqui) [inline src/pure/cmd_analysis.ml:148]: Thanks to remove symqidlist which is the name of a definition and thus should not be part of qidents_of_cmd.

**Comment by fblanqui** [comment]
CI failed because 'make sanity' failed (see https://travis-ci.org/github/Deducteam/lambdapi/jobs/751961536 for instance). Please do 'make sanity_check' and fix the problems (lines too long probably).

File: src/pure/cmd_analysis.ml (hunk 1 above)
Comment (firewall2142) [reply src/pure/cmd_analysis.ml]: done

File: src/pure/cmd_analysis.ml [hunk 2]
```diff
@@ -52,6 +52,48 @@ and qidents_of_p_term (term : Syntax.p_term) =
...
   | Syntax.P_Expl term -> qidents_of_p_term term
 
+and qidents_of_rw_patt (rwpat : Syntax.p_rw_patt) =
+  match rwpat with
+  | Syntax.P_rw_Term pt -> qidents_of_p_term pt
+  | Syntax.P_rw_InTerm pt -> qidents_of_p_term pt
+  | Syntax.P_rw_InIdInTerm (_, pt) -> qidents_of_p_term pt
+  | Syntax.P_rw_IdInTerm (_, pt) -> qidents_of_p_term pt
+  | Syntax.P_rw_TermInIdInTerm (pt1, _, pt2) ->
+      (qidents_of_p_term pt1) @ (qidents_of_p_term pt2)
```
Comment (fblanqui) [inline src/pure/cmd_analysis.ml:62]: code style: parentheses are not necessary here and on line 64 (see line 70).

File: src/pure/cmd_analysis.ml [hunk 3]
```diff
@@ -52,6 +52,48 @@ and qidents_of_p_term (term : Syntax.p_term) =
...
+and qidents_of_p_tactic (term: Syntax.p_tactic) =
+  match term.elt with
+  | Syntax.P_tac_refine pt -> qidents_of_p_term pt
+  | Syntax.P_tac_apply pt -> qidents_of_p_term pt
+  | Syntax.P_tac_rewrite (_, rwpat_lo, pt) ->
+    let rwqids =
+      match rwpat_lo with
+      | Some rwpat -> qidents_of_rw_patt rwpat.elt
+      | None -> []
+    in
```
Comment (fblanqui) [inline src/pure/cmd_analysis.ml:91]: You could factorize these lines, lines 139-143 and line 138 by using List.map_default:
- lines 88-91 replaced by `List.map_default qidents_of_rw_patt [] rwpat.elt`
- lines 140-143 replaced by `List.map_default (List.concat_map qidents_of_p_tactic) [] ptac_list`
- line 138 removed
- some_or_empty replaced by `List.map_default (fun x -> [x]) []`

File: src/pure/cmd_analysis.ml [hunk 4]
```diff
@@ -92,10 +134,17 @@ let qidents_of_cmd (cmd : t) =
   | Syntax.P_require_as (_, _) -> []
   | Syntax.P_open _ -> []
   | Syntax.P_rules rules -> concat_map qidents_of_p_rule rules
-  | Syntax.P_symbol {p_sym_arg;p_sym_typ;p_sym_trm;_} ->
+  | Syntax.P_symbol {p_sym_arg;p_sym_typ;p_sym_trm;p_sym_prf;_} ->
     let some_or_empty = function Some arg -> [arg] | None -> [] in
+    let prfqidlist = begin
```
Comment (fblanqui) [inline src/pure/cmd_analysis.ml:139]: codestyle: begin-end useless here

File: src/pure/cmd_analysis.ml [hunk 5]
```diff
@@ -92,10 +134,17 @@ let qidents_of_cmd (cmd : t) =
...
+      match p_sym_prf with
+      | Some (ptac_list, _) -> List.concat_map qidents_of_p_tactic ptac_list
+      | None -> []
+    end in
     let terms_list = some_or_empty p_sym_typ @ some_or_empty p_sym_trm in
-    filter_bound_qidents p_sym_arg terms_list
+    let res =  (filter_bound_qidents p_sym_arg terms_list) in
```
Comment (fblanqui) [inline src/pure/cmd_analysis.ml:145]: spaces and parentheses useless here, and lines 145-146 can perhaps be put on a single line:
`filter_bound_qidents p_sym_arg terms_list @ prfqidlist`

Comment (firewall2142) [reply src/pure/cmd_analysis.ml]: Done wherever changes were possible

**Comment by fblanqui** [comment]
Thanks @firewall2142 !

**Pull request merged**